pytorch_lightning/trainer/data_loading.py

```diff
@@ -212,14 +213,19 @@ def _reset_eval_dataloader(self, model: LightningModule, mode: str) -> Tuple[int
             dataloaders = [dataloaders]
 
         # shuffling in val and test set is bad practice
+        dataloaders_clean = []
         for loader in dataloaders:
-            if mode in ('val', 'test') and hasattr(loader, 'sampler') and isinstance(loader.sampler, RandomSampler):
-                raise MisconfigurationException(
-                    f'Your {mode}_dataloader has shuffle=True, it is best practice to turn'
-                    ' this off for validation and test dataloaders.')
+            if loader is not None:
+                if mode in ('val', 'test') and hasattr(loader, 'sampler') and isinstance(loader.sampler, RandomSampler):
+                    raise MisconfigurationException(
+                        f'Your {mode}_dataloader has shuffle=True, it is best practice to turn'
+                        ' this off for validation and test dataloaders.')
+                dataloaders_clean.append(loader)
+            else:
```

Comment:
I would keep the previous logic and here add simply:
```
if any([dl is None for dl in dataloaders]):
    rank_zero_warn("One of given dataloaders is None and it will be skipped.")
```

Reply:
ok sure, makes sense